### PR TITLE
[Darjeeling] Instantiate Darjeeling

### PR DESCRIPTION
**Pull request by Razer6**

### CODE REVIEW

File: hw/top_earlgrey/ip_autogen/pinmux/rtl/pinmux.sv
Comment (andreaskurth) [inline hw/top_earlgrey/ip_autogen/pinmux/rtl/pinmux.sv:337]: These lines get moved further up due to a change in the template. No functional change.

(This is just a note; no action required.)

File: hw/ip/prim/lint/prim_ram_1r1w.waiver
Comment (andreaskurth) [inline hw/ip/prim/lint/prim_ram_1r1w.waiver:1]: The commit message says `rst_b_ni` but it should say `rst_a_ni` (and the change is correct)

File: util/topgen/templates/toplevel_pkg.sv.tpl
Comment (andreaskurth) [inline util/topgen/templates/toplevel_pkg.sv.tpl:40]: Please create an issue and reference it here (`TODO(#....)`) so we can track this TODO item

**Review by andreaskurth — COMMENTED** [review]
This PR is huge because files for Darjeeling get autogenerated and added. This is nicely separated into one commit, however, which makes it possible to manually review the other commits.

Overall LGTM, just a few minor suggestions/requests.

Note 1: `hw/top_darjeeling` still has a `FUSESOC_IGNORE` file, which is necessary to not disturb `top_earlgrey` until the FuseSoC update and virtual cores transition has happened. Those working with `top_darjeeling` need to
```
mv hw/top_{darjeeling,earlgrey}/FUSESOC_IGNORE
touch hw/top_englishbreakfast/FUSESOC_IGNORE
```
for now.

Note 2: Top-level simulations (even those without SW) don't yet work because `chip_sim_cfg.hjson` hasn't been added yet. I think it's okay to do this in a follow-up PR, though that could uncover a few more changes necessary to be able to elaborate the top-level design.

Note 3: Some IPs, including the DMA, can already be simulated through Darjeeling's top sim config (`util/dvsim/dvsim.py hw/top_darjeeling/dv/top_darjeeling_sim_cfgs.hjson --select-cfgs dma ...`), while others are still uncommented in `top_darjeeling_sim_cfgs.hjson` and of those some (such as `clkmgr`) don't elaborate yet. Also this, I think, is okay to resolve in a follow-up PR.

Thanks @Razer6!

**Comment by andreaskurth** [comment]
CI checks _Verible lint_ and _Check for and block unauthorized changes_ are failing due to the limitation in number of files. I'll run those checks locally before this PR gets merged.

File: hw/top_darjeeling/top_pkg.core
Comment (Razer6) [reply hw/top_darjeeling/top_pkg.core:9]: Do those core files need to specify a virtual core?

File: hw/ip/prim/lint/prim_ram_1r1w.waiver
Comment (Razer6) [reply hw/ip/prim/lint/prim_ram_1r1w.waiver:1]: Changed the commit message.

File: hw/top_darjeeling/physical_pads.core
Comment (Razer6) [reply hw/top_darjeeling/physical_pads.core:5]: Same as below.

**Comment by Razer6** [comment]
Thanks for the review @andreaskurth.

> Note 2: Top-level simulations (even those without SW) don't yet work because `chip_sim_cfg.hjson` hasn't been added yet. I think it's okay to do this in a follow-up PR, though that could uncover a few more changes necessary to be able to elaborate the top-level design.

I think it's ok for now. Adding the `chip_sim_cfg.hjson` would pull in much more files. I would do that in a separate PR.

> Note 3: Some IPs, including the DMA, can already be simulated through Darjeeling's top sim config (`util/dvsim/dvsim.py hw/top_darjeeling/dv/top_darjeeling_sim_cfgs.hjson --select-cfgs dma ...`), while others are still uncommented in `top_darjeeling_sim_cfgs.hjson` and of those some (such as `clkmgr`) don't elaborate yet. Also this, I think, is okay to resolve in a follow-up PR.

I added all IPs.

File: hw/top_darjeeling/top_pkg.core
Comment (andreaskurth) [reply hw/top_darjeeling/top_pkg.core:9]: Right, with virtual core support this core file would have the name suggested above and specify `lowrisc:constants:top_pkg` as virtual core.

Dependent cores then either have `lowrisc:constants:top_pkg` -- if they don't depend on a _specific_ top -- or `lowrisc:constants:top_darjeeling_top_pkg` -- if they do depend on a specific top -- in their dependencies.

We don't have proper support for virtual cores just yet, so I leave it to you whether you want to do this in this or in a follow-up PR.

Comment (Razer6) [reply hw/top_darjeeling/top_pkg.core:9]: Let's do that (and the other Todo) in a follow up-PR to unblock multitop SW, which depends on that change. CI is green (so far it can), so let's get that in :)

File: util/topgen/templates/toplevel_pkg.sv.tpl
Comment (andreaskurth) [reply util/topgen/templates/toplevel_pkg.sv.tpl:40]: Issue: #25313

Comment (Razer6) [reply util/topgen/templates/toplevel_pkg.sv.tpl:40]: I'll link to that in a follow up PR

**Comment by andreaskurth** [comment]
CHANGE AUTHORIZED: hw/ip/keymgr/rtl/keymgr.sv
CHANGE AUTHORIZED: hw/ip/keymgr/rtl/keymgr_input_checks.sv
CHANGE AUTHORIZED: hw/ip/rv_dm/rtl/rv_dm.sv
CHANGE AUTHORIZED: hw/top_earlgrey/ip_autogen/pinmux/rtl/pinmux.sv

The CI check cannot run on this PR (too many files), but I ran it locally. The files above need to be authorized. None of the changes has a functional impact on Earlgrey.

**Comment by andreaskurth** [comment]
For the Verible lint check, which CI also cannot run due to the file limit, the report includes
```
hw/top_darjeeling/rtl/autogen/top_darjeeling.sv:71:101-106: Line length exceeds max: 100; is: 106 [Style: line-length] [line-length]
```
@Razer6 This is a generated line; would it be simple to insert line breaks?

**Comment by vogelpi** [comment]
CHANGE AUTHORIZED: hw/ip/keymgr/rtl/keymgr.sv
CHANGE AUTHORIZED: hw/ip/keymgr/rtl/keymgr_input_checks.sv
CHANGE AUTHORIZED: hw/ip/rv_dm/rtl/rv_dm.sv
CHANGE AUTHORIZED: hw/top_earlgrey/ip_autogen/pinmux/rtl/pinmux.sv

None of the changes in this PR have a functional impact on Earlgrey. This is fine.

**Review by andreaskurth — APPROVED** [review]

**Pull request merged**
